common/thorhelper/roxiehelper.ipp

```diff
@@ -59,8 +59,25 @@ enum TracingCategory
 class LogItem;
 interface IRoxieContextLogger : extends IContextLogger
 {
-    // Override base interface with versions that add prefix
-    // We could consider moving some or all of these down into IContextLogger
+    virtual StringBuffer &getLogPrefix(StringBuffer &ret) const = 0;
+    virtual bool isIntercepted() const = 0;
+    virtual void CTXLOGa(TracingCategory category, const LogMsgCategory & cat, const LogMsgJobInfo & job, LogMsgCode code, const char *prefix, const char *text) const = 0;
+    void CTXLOGae(IException *E, const char *file, unsigned line, const char *prefix, const char *format, ...) const __attribute__((format(printf, 6, 7)));
+    virtual void CTXLOGaeva(IException *E, const char *file, unsigned line, const char *prefix, const char *format, va_list args) const = 0;
+    virtual void CTXLOGl(LogItem *) const = 0;
+    virtual bool isBlind() const = 0;
+    virtual void gatherStats(CRuntimeStatisticCollection & merged) const = 0;
+    virtual StringBuffer &getStats(StringBuffer &s) const = 0;
+    virtual bool collectingDetailedStatistics() const = 0;
+};
+
+template <class INTERFACE>
+class CRoxieContextLogger : public CDefaultContextLogger<INTERFACE>
```

Comment:
see note below.
I don't think there's any need for this itself to be a template, i.e. can be:
class CRoxieContextLogger : public CDefaultContextLogger<IRoxieContextLogger>